content/2022-01-04-brigade-2.2.0.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Brigade 2.2.0"
+slug: "2022-01-04-brigade-2.2.0"
+description: "Brigade 2.2.0 has been released"
+date: "2022-01-04"
+author: "Kent Rancourt"
+summary: "Brigade 2.2.0 has been released."
+---
+
+Happy New Year, Brigadiers!
+
+We're starting the new year off right with our second minor release of Brigade 2.
+
+v2.2.0 continues the trend started with v2.1.0 by delivering just a handful of small, incremental improvements. Notably, in this release, we've corrected a bug in the user list endpoint and made it easy for script authors to designate jobs as fallible -- that is, _allowed to fail without forcing the overall workflow defined by the script to fail_. This is a great feature for cases where script authors may wish to treat a certain job's failure as informational in nature rather than fatal.
+
+To demonstrate fallible jobs, consider this small fragment of a `brigade.ts` file:
+
+```typescript
+events.on("brigade.sh/cli", "exec", async event => {
+  let job0 = new Job("test-job0", "alpine", event)
+  job0.primaryContainer.command = ["false"]
+
+  let job1 = new Job("test-job1", "alpine", event)
+  job1.primaryContainer.command = ["true"]
+
+  await Job.sequence(job0, job1).run()
+})
+```
+
+In this example, wherein `job0` will always fail, the sequence of `job0` followed by `job1` will never complete and the worker executing this script will _always_ fail. Workarounds exist, but with fallible jobs, this script requires only the slightest modification to ensure `job0` has no bearing on the success or failure of the worker that's executing it:
+
+```typescript
+events.on("brigade.sh/cli", "exec", async event => {
+  let job0 = new Job("test-job0", "alpine", event)
+  job0.primaryContainer.command = ["false"]
+  job0.fallible = true
+
+  let job1 = new Job("test-job1", "alpine", event)
+  job1.primaryContainer.command = ["true"]
+
+  await Job.sequence(job0, job1).run()
+})
+```
+
+At the moment, our favorite practical use of this feature is for running `yarn audit` as part of our CI pipelines on JavaScript and TypeScript projects. While we obviously take vulnerabilities reported by `yarn audit` seriously, we've got [Dependabot](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates) keeping an eye on those and it is frequently the case that a PR that fails `yarn audit` did not modify project dependencies, is not responsible for the failure, and is merely a victim of poor timing. Rather than allow such PRs to get stuck in limbo while unrelated dependency issues are sorted out, we elect to treat the results of `yarn audit` as informational only. The following fragment demonstrates:
```

Comment:
```suggestion
At the moment, our favorite practical use of this feature is for running `yarn audit` as part of our CI pipelines on JavaScript and TypeScript projects. While we obviously take vulnerabilities reported by `yarn audit` seriously, we've got [Dependabot](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates) keeping an eye on those and it is frequently the case that a PR that fails `yarn audit` does not modify project dependencies, is not responsible for the failure, and is merely a victim of poor timing. Rather than allow such PRs to get stuck in limbo while unrelated dependency issues are sorted out, we elect to treat the results of `yarn audit` as informational only. The following fragment demonstrates:
```

Reply:
Nice catch on inconsistent tense here. I'm going to try it like this:

> ... a PR that failed `yarn audit` did not modify project dependencies, was not responsible for the failure, and was merely a victim of poor timing.

Reply:
Fixed in 730c328600d5b536096bd5520998fe059c19b787